graphite/graphite_tcp.go

```diff
@@ -129,6 +129,7 @@ func (t *TCPServer) handleConnection(conn net.Conn) {
 		// Read up to the next newline.
 		buf, err := reader.ReadBytes('\n')
 		if err != nil {
+			flush <- struct{}{}
```

Comment:
I'm not sure this feels right to me.  Exposing the flush channels seems a little raw.  I liked it better when the flush channel was not exposed, and you just called the `Flush()` method.